### PR TITLE
performance improvement for process_cell()

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Increase performances on /turf/open/process_cell() by changing a typed for and making use of `as anything`, plus removed unnecessary proc call

## Why It's Good For The Game
The test has been done in a controlled environment, a 7x7 airless room with one canister of plasma and one canister of oxygen released inside. The profiling ran from the start of the fire to the end of it.
Before any changes:
image
After the `as anything` forloop:
image
After the archive() removal:
image

## Changelog

:cl:
code: increased atmos performances a bit more
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -259,12 +261,12 @@
...
-	for(var/t in adjacent_turfs)
-		var/turf/open/enemy_tile = t
-
+	for(var/turf/open/enemy_tile as anything in adjacent_turfs)
 		if(fire_count <= enemy_tile.current_cycle)
 			continue
-		enemy_tile.archive()
+		enemy_tile.air.archive()
+		enemy_tile.archived_cycle = SSair.times_fired
+		enemy_tile.temperature_archived = enemy_tile.temperature
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:269]: Make this a define and use it everywhere this pattern appears, if we're gonna unroll it we should do it properly

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Good, the as anything is more a code quality thing but I'll bite

**Comment by Ghilker** [comment]
> Good, the as anything is more a code quality thing but I'll bite

From what I've seen, the as anything increased performances a lot

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -141,14 +141,6 @@
 	SEND_SIGNAL(src, COMSIG_TURF_EXPOSE, air, exposed_temperature)
 	check_atmos_process(src, air, exposed_temperature) //Manually do this to avoid needing to use elements, don't want 200 second atom init times
 
-/turf/proc/archive()
-	temperature_archived = temperature
-
-/turf/open/archive()
-	air.archive()
-	archived_cycle = SSair.times_fired
-	temperature_archived = temperature
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:150]: Rather then putting gas_mixture on /turf, keep archive, and unroll it just for the stuff in process_cell, using the define in turf/open/archive()

**Comment by LemonInTheDark** [comment]
With defines is this still good? wanna be sure cause lmao byond

**Comment by Ghilker** [comment]
yeah is roughly the same without the defines

**Comment by LemonInTheDark** [comment]
Profiles look a bit off, would expect slightly higher cost in process_cell with slightly lower totals, but that's a lot a lot and mostly in calling. Gonna a/b test tomorrow, merging pending that

**Comment by Ghilker** [comment]
Nevermind, I didn't rebuild the dm, so it wasn't with the new changes lol, I remake the check and the total CPU was a bit higher, near 4.1 and the calls were around 29k

**Comment by LemonInTheDark** [comment]
> Profiles look a bit off, would expect slightly higher cost in process_cell with slightly lower totals, but that's a lot a lot and mostly in calling. Gonna a/b test tomorrow, merging pending that

"Tomorrow" Good luck with that 4head

Anyway, did some testing.
Redefined process_cell to
```
/turf/open/process_cell(fire_count)
	var/type = rand(1,2)
	switch(type)
		if(1)
			process_cell_old(fire_count)
		if(2)
			process_cell_base(fire_count)
```
image
Self is higher, which is expected, total for _base actually kinda fluctuates, it's sometimes a point or two lower then _old, though it's kinda hard to tell because the cost of archive is such a small part of the cost of process_cell.
Seems alright though.

It's totally a micro op, but the define is somehow more clear then the proc name so I'm having a hard time caring

**Pull request merged**